rafttest/interaction_env_handler_stabilize.go

```diff
@@ -25,7 +26,26 @@ import (
 )
 
 func (env *InteractionEnv) handleStabilize(t *testing.T, d datadriven.TestData) error {
-	idxs := nodeIdxs(t, d)
+	idxs := nodeIdxs(t, d) // skips key=value args
+	for _, arg := range d.CmdArgs {
+		for i := range arg.Vals {
+			switch arg.Key {
+			case "log-level":
+				var level string
+				arg.Scan(t, i, &level)
+				old := env.Output.Lvl
+				defer func() {
+					env.Output.Lvl = old
+					if level == "none" {
+						env.Output.Builder = &strings.Builder{}
+					}
```

Comment:
Could you clarify why do you reset the env.Output.Builder when level == "none"? If we really need to do it, we should get the details encapsulated in `(*InteractionEnv) LogLevel`.
```suggestion
					_ = env.LogLevel(old)
```

Reply:
This is because we not only want to hide the log output, we also want to hide the test output. Consider this example:

```
stabilize
----
> 2 handling Ready
  Ready MustSync=false:
  Lead:0 State:StatePreCandidate
  Messages:
  2->1 MsgPreVote Term:2 Log:1/12
  2->3 MsgPreVote Term:2 Log:1/12
  INFO 2 received MsgPreVoteResp from 2 at term 1
  INFO 2 has received 1 MsgPreVoteResp votes and 0 vote rejections
```

If we only set `env.LogLevel(none)` then it will hide the lines starting with `INFO`, but still output the other lines which are not emitted by the logger but rather the test framework.

This is the same logic as with the `log-level` command, which happens here:

https://github.com/etcd-io/raft/blob/768abf8a1c58ebd9dd859c71b3fba6704400bc82/rafttest/interaction_env_handler.go#L201-L206

Reply:
I went a bit further, and made the `Output.Write*` methods respect the `NONE` log level. This seemed cleaner, and will only suppress output for the duration that `Lvl` was actually set to `NONE`.

One minor effect is that we don't easily know whether we suppressed any output, so we now always output `ok (quiet)` with `NONE`. That seems fine.